drivers/firmware/scmi/nxp/shmem.c

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/crc.h>
+#include "../shmem_internal.h"
+
+LOG_MODULE_REGISTER(arm_scmi_shmem_nxp);
+
+int scmi_shmem_vendor_validate_message(const struct scmi_shmem_layout *layout)
```

Comment:
maybe we can use a more generic name for these functions? e.g: `scmi_shmem_vendor_write_message` and `scmi_shmem_vendor_read_message`. We don't really know what other vendors do with the implementation-defined bits.